doc/src/manual/types.md

```diff
@@ -1590,15 +1590,71 @@ to adjust printing.
+  By default (if no 3-argument method is defined for `typeof(x)`),
+  it calls the 2-argument `show(io, x)`.
+
+* [`print(io, x)`](@ref) by default calls `show(io, x)`,
+  but a few types have a distinct `print` format —
+  most notably, when `x` is a string, `print` outputs the raw text
+  whereas `show` outputs an escaped string enclosed in quotation marks.
+  ([`println`](@ref) and [`printstyled`](@ref) both call [`print`](@ref),
+  but add a newline or styling, respectively.)
+
+* [`display(x)`](@ref) tells the current environment to display `x`
+  in whatever way it thinks best.
+  This is the function used by the REPL to output the result of an evaluated expression.
+  In the REPL, `display` calls `show(io, MIME"text/plain", x)`.
+  In a graphical environment, such as Jupyter or Pluto,
+  `display` might prefer a non-plaintext representation of an object
+  (such as HTML, Markdown, or a PNG or SVG image), if the corresponding 3-argument `show`
+  method is defined (as determined by [`showable`](@ref)).
+
+* [`write(io, x)`](@ref), if it is defined
+  (it generally has *no* default definition for new types),
+  writes a "raw" binary representation of `x` to `io`,
+  e.g. an `x::Int32` will be written as 4 bytes.
+
+The `io` argument for all the above functions is optional
+and defaults to [`stdout`](@ref) if it is omitted;
+this is the default output stream, typically a terminal window.
+(`display` has no `io` argument
+because the output format and stream/device is chosen by the display backend.)
+
+More generally, `io` (of type [`IO`](@ref)) specifies a desired output stream
+(such as a file, buffer, or pipe).
+In the REPL, `io` is an [`IOContext`](@ref) wrapper around `stdout`, as described below.
+It is also helpful to be familiar with the metadata that can be attached to an `io` stream
+by an [`IOContext`](@ref) wrapper.
+For example,
+the REPL sets the `:limit => true` flag from `display` for an evaluated expression,
+in order to limit the output to fit in the terminal;
+you can query this flag with `get(io, :limit, false)`.
+And when displaying an object contained within, for example, a multi-column matrix,
+the `:compact => true` flag could be set,
+which you can query with `get(io, :compact, false)`.
+
+The functions `string`, `annotatedstring`, and `repr`
+may be used to save values as a string (rather than output to `io`).
+These functions call either `show` or `print` as indicated below,
+so it is usually not necessary to define new methods for them separately.
+(Note that [`convert(String, x)`](@ref) is not defined for most `x`
+since the `convert` function is intended for implicit conversions.)
+
+* [`string(x)`](@ref) and [`annotatedstring(x)`](@ref) call [`print(io, x)`](@ref).
+* [`repr(x)`](@ref), with one argument, calls the 2-argument [`show(io, x)`](@ref).
+* `repr("text/plain", x)`, with two arguments, calls the 3-argument [`show(io, ::MIME"text/plain", x)`](@ref).
```

Comment:
Can this be generalized to `repr(mime, x)` $\rightarrow$ `show(io, mime, x)`?
I also do not understand the change of form from `"text/plain"` to `::MIME"text/plain"`. Does either option work in either location?

Reply:
Yes, it can be generalized I think. 

(Writing from my phone so I can’t double check)